host-agent/internal/realtime/client.go

```
@@ -0,0 +1,271 @@
+}
+
+func New(config Config) *Client {
+	return &Client{config: config, executor: executor.New(), commands: make(chan struct{}, 4)}
+}
+
+func (c *Client) Run(ctx context.Context) error {
+	backoff := time.Second
+	for ctx.Err() == nil {
+		err := c.runConnection(ctx)
+		if ctx.Err() != nil {
+			return nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 78) but it returns nil

(nilerr)